docs/tutorials/03-Logging-particles-undergoing-detection/Logging-particles-example.ipynb

```diff
@@ -0,0 +1,228 @@
+    "purpose. In this example, we will demonstrate how to use the logging functionality to\n",
+    "identify and track particles that are undergoing nucleation effectively identifying\n",
+    "particles belonging to the critical nucleus. The render function in the next (hidden)\n",
+    "cell will render the snapshot at the detected change point using fresnel. You can find\n",
+    "the source code for this function on github."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 57,
+   "metadata": {
+    "jupyter": {
+     "source_hidden": true
+    },
+    "nbsphinx": "hidden",
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import math\n",
+    "\n",
+    "import fresnel\n",
+    "import IPython\n",
+    "import numpy as np\n",
+    "\n",
+    "device = fresnel.Device()\n",
+    "tracer = fresnel.tracer.Path(device=device, w=300, h=300)\n",
+    "\n",
+    "\n",
+    "def render(snapshot):\n",
+    "    L = snapshot.configuration.box[0]\n",
+    "    scene = fresnel.Scene(device)\n",
+    "    geometry_B = fresnel.geometry.Sphere(\n",
+    "        scene, N=np.sum(snapshot.particles.typeid == 1), radius=0.5\n",
+    "    )\n",
+    "    geometry_B.material = fresnel.material.Material(\n",
+    "        color=fresnel.color.linear([1 / 255, 109 / 255, 1 / 255]), roughness=0.5\n",
+    "    )\n",
+    "    geometry = fresnel.geometry.Sphere(\n",
+    "        scene, N=np.sum(snapshot.particles.typeid == 0), radius=0.125\n",
+    "    )\n",
+    "    geometry.material = fresnel.material.Material(\n",
+    "        color=fresnel.color.linear([252 / 255, 209 / 255, 1 / 255]),\n",
+    "        roughness=5,\n",
+    "        metal=0,\n",
+    "        specular=0.95,\n",
+    "        spec_trans=0.95,\n",
+    "    )\n",
+    "    geometry.position[:] = snapshot.particles.position[\n",
+    "        snapshot.particles.typeid == 0, :\n",
+    "    ]\n",
+    "    geometry.outline_width = 0.04\n",
+    "    geometry_B.position[:] = snapshot.particles.position[\n",
+    "        snapshot.particles.typeid == 1, :\n",
+    "    ]\n",
+    "    geometry_B.outline_width = 0.04\n",
+    "    fresnel.geometry.Box(scene, [L, L, L, 0, 0, 0], box_radius=0.02)\n",
+    "\n",
+    "    scene.lights = [\n",
+    "        fresnel.light.Light(\n",
+    "            direction=(0, 0, 1), color=(0.8, 0.8, 0.8), theta=math.pi\n",
+    "        ),\n",
+    "        fresnel.light.Light(\n",
+    "            direction=(1, 1, 1), color=(1.1, 1.1, 1.1), theta=math.pi / 3\n",
+    "        ),\n",
+    "    ]\n",
+    "    scene.camera = fresnel.camera.Orthographic(\n",
+    "        position=(L * 2, L, L * 2),\n",
+    "        look_at=(0, 0, 0),\n",
+    "        up=(0, 1, 0),\n",
+    "        height=L * 1.4 + 1,\n",
+    "    )\n",
+    "    scene.background_alpha = 1\n",
+    "    scene.background_color = (1, 1, 1)\n",
+    "    return IPython.display.Image(tracer.sample(scene, samples=500)._repr_png_())"
```

Comment:
```suggestion
    "    return IPython.display.Image(tracer.sample(scene, samples=2500)._repr_png_())"
```

If we are only rendering this once per doc build, we may as well output a cleaner image.